src/io/read.rs

```diff
@@ -26,24 +38,89 @@ pub trait Read {
 
     /// Returns the number of bytes still available to read.
     fn remaining_data(&self) -> usize;
+}
+assert_obj_safe!(Read);
 
+/// Convenience functions for reading integers from a [`Read`].
+pub trait ReadInt: Read {
     /// Reads a little-endian integer.
     ///
     /// # Note
     /// Do not implement this function yourself. Callers are not required to
     /// call it in order to actually perform a read, so whether or not it is
     /// called is an implementation detail.
     #[inline]
-    fn read_le<I: LeInt>(&mut self) -> Result<I, io::Error>
-    where
-        Self: Sized,
-    {
+    fn read_le<I: LeInt>(&mut self) -> Result<I, io::Error> {
         I::read_from(self)
     }
 }
-assert_obj_safe!(Read);
 
-impl<R: Read + ?Sized> Read for &'_ mut R {
+impl<R: Read + ?Sized> ReadInt for R {}
+
+/// A [`Read`] that can zero-copy read data of a particular lifetime.
+///
+/// Most implementations can get away with `impl ReadZero<'_> for MyReader {}`.
+///
+/// # Safety
+///
+/// Buffers returned by `read_direct` must observe the alignment requirements
+/// requested of them. The default implementation does this automatically.
```

Comment:
"the alignment requirements requested of them" is vague. How about "Buffers returned by `read_direct` must have the size and alignment specified by `layout`".